Node JS Server, and Styled Components

DylanNeil opened the pull request:
This branch actually includes the changes made in the nodeServer branch. I forgot to checkout again from master, so this branch was built on top of nodeServer branch. I will delete nodeServer.

- Styled Components
   - installed Styled Components
   - Created button, and a variation of that button to test
       - Made changes in theme.js to set material UI button
       - Added simpleButton.js using Styled Components to create a variation of the button from theme.js

- Node Server
   - Includes the work we have done together. I would just feel better if everything was saved in master at this point, since the inclusion of our updates doesn't currently effect anything on the client side.

fawcilize commented:
I added a commit to remove node modules with the following commands:
1. `git rm -r --cached .` which removed the node_modules folder
2. Changed `/node_modules` to `node_modules` (so that it ignores recursively instead of just the root)
3. `git add .`
4. `git push`

You'll want to squash this pull request from the dropdown.
![image](https://user-images.githubusercontent.com/16828669/93596261-051e6280-f987-11ea-8bef-a7b951c4d89e.png)

Otherwise, you'll be temporarily including all those extraneous files in the git history, which is downloaded every time someone clones the repository. If you don't want to squash your commits, then we'll have to do some additional logic to change your history before we merge it.

fawcilize commented on server/closure.js:
```diff
@@ -0,0 +1,22 @@
+function test() {
```
I think it's great to save this file, but perhaps they should be collected in a separate folder

fawcilize commented on server/server.js:
```diff
@@ -0,0 +1,44 @@
+});
+
+app.delete('/events/:id',(req, res) => {
+  const requestId = events.find((event) => event.id === req.params.id);
```
You're finding the event, but it doesn't look like you're actually deleting the event.

fawcilize commented on server/server.js:
```diff
@@ -0,0 +1,44 @@
+
+app.delete('/events/:id',(req, res) => {
+  const requestId = events.find((event) => event.id === req.params.id);
+  res.status(202).send(requestId);
```
The standard response for delete calls is a status code of `204`, No Content, with an empty response body.

fawcilize commented on server/server.js:
```diff
@@ -0,0 +1,44 @@
+
+// create a GET route
+app.get('/events/:id', (req, res) => {
+    const event = events.find((event) => event.id === req.params.id);
```
We need to return a status code of `404`, Not Found, in the case that the event is not in the list.

fawcilize replied:
Challenge Question:
Imagine if this events list had a million entries. It would be awfully slow to traverse, which would cause an appearance of lag because of the slow response time. Do you know how we can make this call have a consistent response time, regardless of the number of events in the list?

PS: It's not worth spending too much time on this. The end goal is to put these events in a database, and not in memory, so whatever solution we have here will be deleted.

We'll go over it the next time we meet.

Hint: Look up big O notation

fawcilize reviewed (approved):
I've added labels linking the issues to this pull request. It's a good practice to keep the project board up to date with what you're working on. It'll save you a lot of headache later.

DylanNeil commented:
Challenge Question:

The best way to do this would be with logarithmic time, notated as
O(log(n)), in which n is the data or amount of data. The idea is to sort
the data and split it in half. Check if the value is higher or lower then
the center, cut off the invalid half, then repeat the process with the
remaining half until the targeted data is found.

On Fri, Sep 18, 2020 at 8:33 AM Kevin Fawcett <notifications@github.com>
wrote:

> *@fawcilize* approved this pull request.
>
> I've added labels linking the issues to this pull request. It's a good
> practice to keep the project board up to date with what you're working on.
> It'll save you a lot of headache later.
> ------------------------------
>
> In server/closure.js
> <https://github.com/DylanNeil/ReactProject/pull/45#discussion_r490910117>:
>
> > @@ -0,0 +1,22 @@
> +function test() {
>
> I think it's great to save this file, but perhaps they should be collected
> in a separate folder
> ------------------------------
>
> In server/server.js
> <https://github.com/DylanNeil/ReactProject/pull/45#discussion_r490911072>:
>
> > +
> +app.use(bodyParser.json());
> +
> +app.post('/events', function test2(req,res) {
> +  idCounter += 1;
> +  const eventObject = {
> +    id: idCounter,
> +    name: req.body.name
> +  };
> +  events.push(eventObject);
> +  console.log(events);
> +  res.status(201).send({ eventObject });
> +});
> +
> +app.delete('/events/:id',(req, res) => {
> +  const requestId = events.find((event) => event.id === req.params.id);
>
> You're finding the event, but it doesn't look like you're actually
> deleting the event.
> ------------------------------
>
> In server/server.js
> <https://github.com/DylanNeil/ReactProject/pull/45#discussion_r490911455>:
>
> > +app.use(bodyParser.json());
> +
> +app.post('/events', function test2(req,res) {
> +  idCounter += 1;
> +  const eventObject = {
> +    id: idCounter,
> +    name: req.body.name
> +  };
> +  events.push(eventObject);
> +  console.log(events);
> +  res.status(201).send({ eventObject });
> +});
> +
> +app.delete('/events/:id',(req, res) => {
> +  const requestId = events.find((event) => event.id === req.params.id);
> +  res.status(202).send(requestId);
>
> The standard response for delete calls is a status code of 204, No
> Content, with an empty response body.
> ------------------------------
>
> In server/server.js
> <https://github.com/DylanNeil/ReactProject/pull/45#discussion_r490912470>:
>
> > +    name: req.body.name
> +  };
> +  events.push(eventObject);
> +  console.log(events);
> +  res.status(201).send({ eventObject });
> +});
> +
> +app.delete('/events/:id',(req, res) => {
> +  const requestId = events.find((event) => event.id === req.params.id);
> +  res.status(202).send(requestId);
> +    return;
> +});
> +
> +// create a GET route
> +app.get('/events/:id', (req, res) => {
> +    const event = events.find((event) => event.id === req.params.id);
>
> We need to return a status code of 404, Not Found, in the case that the
> event is not in the list.
> ------------------------------
>
> In server/server.js
> <https://github.com/DylanNeil/ReactProject/pull/45#discussion_r490915769>:
>
> > +    name: req.body.name
> +  };
> +  events.push(eventObject);
> +  console.log(events);
> +  res.status(201).send({ eventObject });
> +});
> +
> +app.delete('/events/:id',(req, res) => {
> +  const requestId = events.find((event) => event.id === req.params.id);
> +  res.status(202).send(requestId);
> +    return;
> +});
> +
> +// create a GET route
> +app.get('/events/:id', (req, res) => {
> +    const event = events.find((event) => event.id === req.params.id);
>
> Challenge Question:
> Imagine if this events list had a million entries. It would be awfully
> slow to traverse, which would cause an appearance of lag because of the
> slow response time. Do you know how we can make this call have a consistent
> response time, regardless of the number of events in the list?
>
> PS: It's not worth spending too much time on this. The end goal is to put
> these events in a database, and not in memory, so whatever solution we have
> here will be deleted.
>
> We'll go over it the next time we meet.
>
> Hint: Look up big O notation
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/DylanNeil/ReactProject/pull/45#pullrequestreview-491413037>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACUK5ELRYO5WUUKDWHCR5GDSGNHSPANCNFSM4RCMB34Q>
> .
>
